### PR TITLE
Mayh 5751 upload index

### PR DESCRIPTION
**Pull request by vashu1**
Binary index format, unaligned.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -49,19 +46,17 @@ serviceMain opt = do
...
     runConduit $
       mvarSource submissionReady
-      .| tap (L.catMaybes .| Srv.handleStream (opt ^. optXmlIndexBucket))
-      .| effect updateOffsetMap
-      .| everyNSeconds (opt ^. optKafkaConsumerCommitPeriodSec)  -- only commit ever N seconds, so we don't hammer Kafka.
-      .| effect (\_ -> get >>= logInfo . show . _filesCount)
-      .| commitStoredOffsetsSink consumer
+        .| tap (L.catMaybes .| Srv.handleStream (opt ^. optXmlIndexBucket))
+        .| effect (\_ -> get >>= logInfo . show . _filesCount)
+        .| mvarSink undefined
```
Comment (newhoggy) [inline app/Main.hs]: There's still an `undefined` here.  You need to set it to an actual MVar, I think

File: app/Service.hs [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
-handleStream sr =
+handleStream bucketName =
   L.map crValue
   .| effect (\x -> logInfo $ "[Handle] " <> show (S.submissionFile x))
-  .| L.sinkNull
+  .| L.mapM indexXmlFileEither
+  .| L.mapM (uploadResultToBucket bucketName)
+  .| L.mapM handleErrors
+  .| L.mapM submitProcessedMetric
+  .| L.mapM_ (\_ -> filesCount += 1)
```
Comment (newhoggy) [inline app/Service.hs]: Use `.| L.mapM_ (const (filesCount += 1))`

File: src/App/IndexFile.hs [hunk 3]
```diff
@@ -1,74 +1,90 @@
...
+  where f bs w = bs <> encode w
+
+xmlIndex :: String -> BS.ByteString -> XmlIndex
+xmlIndex path bs = XmlIndex xmlIndexVersion path interests' balancedParens'
   where
-    interests' = indexVectorToString interestsVector
-    balancedParens' = indexVectorToString balancedParensVector
-    xmlSettings = useShortEmptyTags (const False) defaultConfigPP
-    indexElem = unode "index" $ XmlIndex xmlIndexVersion interests' balancedParens'
+    XmlCursor _ (BitShown ib) (SimpleBalancedParens bp) _ = fromByteString bs :: SimpleCursor
```
Comment (jianwan2016) [inline src/App/IndexFile.hs]: Are we still using xml format for the index?

Comment (vashu1) [inline src/App/IndexFile.hs]: nope, it's binary

Comment (jianwan2016) [inline src/App/IndexFile.hs]: So this `vectorToStrictBS` could be removed?

File: app/Main.hs [hunk 4]
```diff
@@ -49,19 +47,17 @@ serviceMain opt = do
...
     runConduit $
       mvarSource submissionReady
-      .| tap (L.catMaybes .| Srv.handleStream (opt ^. optXmlIndexBucket))
-      .| effect updateOffsetMap
-      .| everyNSeconds (opt ^. optKafkaConsumerCommitPeriodSec)  -- only commit ever N seconds, so we don't hammer Kafka.
-      .| effect (\_ -> get >>= logInfo . show . _filesCount)
-      .| commitStoredOffsetsSink consumer
+        .| tap (L.catMaybes .| Srv.handleStream (opt ^. optXmlIndexBucket))
+        .| effect (\_ -> get >>= logInfo . show . _filesCount)
+        .| mvarSink emptyVar
```
Comment (AlexeyRaga) [inline app/Main.hs]: You have removed all the offsets committing stuff and sinking into an empty MVar instead? Why so? ;)

Comment (vashu1) [inline app/Main.hs]: that was long ago, when I thought that it commits offsets for target kafka )

pushed fix

**Comment by AlexeyRaga** [comment]
Looks good, although I was not following this project

**Pull request merged**